build-ps/test-build.sh

```diff
@@ -1,61 +1,85 @@
+        ldd "$elf"
     done
     return
 }
 
 function prepare {
     CURDIR=$(pwd)
+    TMP_DIR="$CURDIR/temp"
+
+    mkdir -p "$CURDIR"/temp
+    mkdir -p "$TMP_DIR"/dbdeployer/tarball "$TMP_DIR"/dbdeployer/deployment
+
+    TARBALLS=""
+    for tarball in $(find . -name "*.tar.gz"); do
+        TARBALLS+=" $(basename $tarball)"
+    done
 
-    mkdir -p $CURDIR/temp
-    TMP_DIR=$CURDIR/temp
-    
-    TARBALL=$(basename $(find . -name "*glibc2.12.tar.gz" | sort))
+    DIRLIST="bin lib lib/private lib/plugin lib/mysqlrouter/plugin lib/mysqlrouter/private"
 }
 
 function install_deps {
     if [ -f /etc/redhat-release ]; then 
-        yum install -y perl-Time-HiRes perl numactl numactl-libs libaio libidn || true
+        yum install -y wget perl-Time-HiRes perl numactl numactl-libs libaio libidn || true
     else
-        apt install -y perl numactl libaio-dev libidn11 || true
+        apt install -y wget perl numactl libaio-dev libidn11 || true
     fi
+    wget -c https://github.com/datacharmer/dbdeployer/releases/download/v1.52.0/dbdeployer-1.52.0.linux.tar.gz -O - | tar -xz
+    mv dbdeployer*.linux /usr/local/bin/dbdeployer
+    export PATH=$PATH:/usr/local/bin
 }
 
 main () {
-
-    DIRLIST="bin lib lib/private lib/plugin lib/mysqlrouter/plugin lib/mysqlrouter/private"
-
     prepare
 
-    echo "Unpacking tarball"
-    cd $TMP_DIR
-    tar xf $CURDIR/$TARBALL
-    cd ${TARBALL%.tar.gz}
+    for tarfile in $TARBALLS; do
+        echo "Unpacking tarball: $tarfile"
+        cd "$TMP_DIR"
+        tar xf "$CURDIR/$tarfile"
+        if [[ "$tarfile" == *minimal* ]]; then
+            cd "${tarfile%-minimal.tar.gz}"
+        else
+            cd "${tarfile%.tar.gz}"
+        fi
 
-    echo "Checking ELFs for not found"
-    for DIR in $DIRLIST; do
-        check_libs $DIR | tee -a $TMP_DIR/libs_err.log
-    done
+        echo "Checking ELFs for not found"
+        for DIR in $DIRLIST; do
+            check_libs "$DIR" >> "$TMP_DIR"/libs_err.log
+        done
 
-    if [[ ! -z $(cat $TMP_DIR/libs_err.log | grep "not found") ]]; then
-        echo "ERROR: There are missing libraries: "
-        cat $TMP_DIR/libs_err.log
-        exit 1
-    fi
+        if [[ ! -z "$(grep \"not found\" $TMP_DIR/libs_err.log)" ]]; then
+            echo "ERROR: There are missing libraries: "
+            grep "not found" "$TMP_DIR"/libs_err.log
+            exit 1
+        fi
+
+        echo "Checking ELFs for any other error"
+        for DIR in $DIRLIST; do
+            if ! check_libs "$DIR"; then
+                exit 1
+            fi
+        done
 
-    # Run MTRs
-    cd mysql-test
-    MTR_BUILD_THREAD=auto ./mtr \
-            --force \
-            --max-test-fail=0 \
-            main.1st 
-    MTR_BUILD_THREAD=auto ./mtr \
-            --force \
-            --suite=auth_sec
+        echo "Invoking dbdeployer to make a test run"
+        dbdeployer unpack --sandbox-binary="$TMP_DIR"/dbdeployer/tarball --prefix=ps "$CURDIR/$TARBALL"
```

Comment:
it will fail due to difference in tarball and directory name. please also fix directory name creation for minimal tarball